video_xblock/video_xblock.py

```diff
@@ -209,3 +255,38 @@ def get_player(self):
         """
         player = BaseVideoPlayer.load_class(self.player_name)
         return player()
+
+    @XBlock.json_handler
+    def save_file(self, request, suffix=''):
+        """
+         Save file either local storage or Amazon S3 bucket and save a path to that file
+
+        if type of storage is s3fs: we get link with GET params, e.g
+            https://.../df924b40b8ac11e6a142080027880ca6.png?Signature=ZoKUlcIqfZIdSu1QvUmVe%2F9ly%2BE%3D&Expires=1480696395&AWSAccessKeyId=AKIAIVZILNCNLO4EDHRA
+        we have to cut it to avoid expire data in request
+
+        file_string:
+            sample of string which we got from ajax -
+            'data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAQCAwMDAgQDAwMEBAQEBQkGBQUFBQsICAYJDQsNDQ0LD...'
+
+        """
+        file_string = request.get('file')
+        if file_string:
+            file_suffix = file_string.split(';')[0].split('/')[-1]
+            filename = uuid1().hex
+            full_filename = '{}.{}'.format(filename, file_suffix)
+            with self.handout.open(full_filename, 'wb') as handout_file:
+                handout_file.write(base64.b64decode(file_string.split(',')[1]))
+                handout_file.close()
+            handout_file_path = self.handout.get_url(full_filename)
```

Comment:
you can set a timeout as kwargs in seconds. For example, for a day. I believe it'd be enough. @z4y4ts What do you think?